test/factorization/cholesky_kernels.cpp

```diff
@@ -125,6 +155,52 @@ using Types = gko::test::cartesian_type_product_t<gko::test::RealValueTypesBase,
 TYPED_TEST_SUITE(CholeskySymbolic, Types, PairTypenameNameGenerator);
 
 
+TYPED_TEST(CholeskySymbolic, KernelComputeSkeletonTree)
+{
+    using matrix_type = typename TestFixture::matrix_type;
+    using index_type = typename TestFixture::index_type;
+    using elimination_forest = typename TestFixture::elimination_forest;
+    for (const auto& pair : this->matrices) {
+        SCOPED_TRACE(pair.first);
+        const auto& mtx = pair.second;
+        // check for correctness: is mtx symmetric?
+        GKO_ASSERT_MTX_EQ_SPARSITY(mtx, gko::as<matrix_type>(mtx->transpose()));
+        const auto dmtx = gko::clone(this->exec, mtx);
+        const auto size = mtx->get_size();
+        const auto skeleton = matrix_type::create(this->ref, size, size[0]);
+        const auto dskeleton = matrix_type::create(this->exec, size, size[0]);
+
+        gko::kernels::reference::elimination_forest::compute_skeleton_tree(
+            this->ref, mtx->get_const_row_ptrs(), mtx->get_const_col_idxs(),
+            size[0], skeleton->get_row_ptrs(), skeleton->get_col_idxs());
+        gko::kernels::GKO_DEVICE_NAMESPACE::elimination_forest::
+            compute_skeleton_tree(this->exec, dmtx->get_const_row_ptrs(),
+                                  dmtx->get_const_col_idxs(), size[0],
+                                  dskeleton->get_row_ptrs(),
+                                  dskeleton->get_col_idxs());
+
+        // check that the created graphs are trees and have the same edge sum
+        const auto weight_sum = this->check_mst(skeleton);
+        const auto dweight_sum = this->check_mst(dskeleton);
+        ASSERT_EQ(weight_sum, dweight_sum);
+        // while the MSTs may be different, they should produce the same
+        // elimination forest as the original matrix
```

Comment:
I think it should still be affected by rounding error because the accumulation order might be different from two minimal spanning trees.

Reply:
the weights are integers, so there is no rounding taking place

Reply:
you are right